### PR TITLE
Faster way to look for snow around pine sapling

### PR DESCRIPTION
**Pull request by tenplus1**
Instead of using voxelmanip to check for snow around a pine sapling this is a lot simpler and faster.

### CODE REVIEW

**Comment by t4im** [comment]
holy cobble, wow! nice catch

Your solution has the additional benefit, that it is less likely to forceload additional areas in some situations.

File: mods/default/trees.lua [hunk 1]
```diff
@@ -27,32 +27,12 @@ end
...
-		for xx = x - 1, x + 1 do
-			local nodid = data[vi]
-			if nodid == c_snow or nodid == c_snowblock or nodid == c_dirtsnow then
-				return true
-			end
-			vi  = vi + 1
-		end
-	end
+	if #minetest.find_nodes_in_area(
+		{x = pos.x - 1, y = pos.y - 1, z = pos.z - 1},
```
Comment (SmallJoker) [inline mods/default/trees.lua]: How about using the builtin functions? This `vector.subtract(pos, 1), vector.add(pos, 1)` looks better IMO.

**Comment by sofar** [comment]
:+1: , either with or without smalljoker's comment addressed.

**Comment by 0-afflatus** [comment]
If that's faster than a voxelmanip then it considerably simplifies the code. 
Does using the builtin functions actually provide any advantage?

**Comment by paramat** [comment]
:+1: 
Oops, the snow check used to be part of pine tree generation so used to use the already initiated voxelmanip, making it fast. When i moved snow check to a seperate function i forgot to use the C++ function instead.
The vvector helper is lua so isn't faster, and this is clearer so it's okay as it is.

**Comment by tenplus1** [comment]
@paramat - I tested using a 1000 for loop to check for snowy nodes around sapling which worked fine and a lot quicker than using voxelmanip.  Will return true so that a snowy pine grows when these nodes are found.

**Comment by paramat** [comment]
Will merge soon.

**Comment by paramat** [comment]
Would this work? It's shorter:
```return #minetest.find_nodes_in_area(...) > 0```

**Comment by paramat** [comment]
This works, checked:
```
-- 'is snow nearby' function

local function is_snow_nearby(pos)
	return #minetest.find_nodes_in_area(
		{x = pos.x - 1, y = pos.y - 1, z = pos.z - 1},
		{x = pos.x + 1, y = pos.y + 1, z = pos.z + 1},
		{"default:snow", "default:snowblock", "default:dirt_with_snow"}) > 0
end
```

**Comment by t4im** [comment]
If that function is called very often, then localizing the `{"default:snow", "default:snowblock", "default:dirt_with_snow"}` could be worth it too.

**Comment by paramat** [comment]
Not worth it, it's only used when saplings grow.

**Comment by paramat** [comment]
f1f96dbe6b2af33f669283252861e1cb74eef496

**Pull request closed without merging**